samples/boards/stm32/pm_lptim/README.rst

```diff
@@ -0,0 +1,47 @@
+.. _Power-Management-LP-Timer:
+
+Power Management LP Timer
+#########################
+
+Overview
+********
+
+This is a simple application which enter/exit the low power mode.
+The led is ON in running mode and is off during low power periods.
+
+.. _Power-Management-LP-Timer-requirements:
+
+Requirements
+************
+
+You will see this error if you try to build sample for an unsupported board:
+
+.. code-block:: none
+
+   Unsupported board: led0 devicetree alias is not defined
+   Unsupported board: sw0 devicetree alias is not defined
+
+The board must have a user button and a LED connected via GPIO pins. These are called "User
+LEDs" on many of Zephyr's :ref:`boards`. The Button and LED must be configured using the
+``sw0`` :ref:`devicetree <dt-guide>` alias and
+``led0`` :ref:`devicetree <dt-guide>` alias.
+This is usually done in the :ref:`BOARD.dts file <devicetree-in-out-files>` or a :ref:`devicetree overlay
+<set-devicetree-overlays>`.
+
+Building and Running
+********************
+
+Build and flash sample as follows, changing ``nucleo_l476rg`` for your board:
+
+.. zephyr-app-commands::
+   :zephyr-app: samples/boards/stm32/pm_lptim
+   :board: nucleo_l476rg
+   :goals: build flash
+   :compact:
+
+After flashing, the LED is on, waiting for entering the low Power mode.
+After half-second delay, the kernel is entering the low power mode (stm32 stop mode),
+for four seconds. The led is off during this period.
+After four seconds, the timeout wake the system and turn the led on again.
+When pressing the user button, the system is also able to exit from low power mode,
+(led on) because the button is configured in interrupt mode on its EXTI line.
```

Comment:
It's unclear from this description and running this exactly how this works.

It appears that the cycle is "turn LED on and busy-wait 0.5 sec, then turn LED off and sleep 4 seconds".  Pressing the button at any time turns LED on, and it gets turned off at the end of the next (or current) busy-wait.

So pressing the button does nothing but affect the LED.  I guess.  I suppose we'd have to measure power to see whether there's an effect if the button is pressed 3.5 seconds into the sleep cycle.

Reply:
I can confirm that the expected behaviour you described, indeed is the case. During my testing (on stm32l1_disco) I've observed that the LED will turn on again after pressing the button, but the true link between the LED and the sleep state seems to be missing. 
I've observed that after the button press it does seem to wake up the SoC and kernel, but after that event the LED can still be on even though the SoC is (supposedly) in sleep mode again. In debug I can see that it went through the SoCs power.c file, and the steps for the sleep mode have been taken, but I haven't confirmed whether it actually is in the sleep mode with a power consumption measurement yet. 

I think it would make more sense to see if it's possible to link the current power_state to the LED's output